### ISSUE TITLE
Go To Definition with the cursor inside the target definition triggers Go To References instead

### ISSUE THREAD

**Issue by mooware** [microsoft/vscode#110060]
We're implementing a custom language server and encountered the following issue:

When using Go To Definition and the position is already inside the definition we would go to, i.e. when something inside the definition refers to itself, VS Code will not jump to the beginning of the definition and highlight it, as it would usually do, but instead will act as if Go To References was used.

I'm assuming this is not intentional, since I don't see why it would act like that. I would expect it to jump to the beginning of the definition and highlight it, even when we're currently inside that range.

To reproduce the issue, I've also tried it with a publicly available language server, specifically the Python extension by Microsoft.

```python
class Foo:
    def foo(self):
        x = Foo()

class Bar:
    def bar(self):
        y = Foo()
```

When using Go To Definition on "Foo" in "y = Foo()", it works as expected. When using it on "x = Foo()", it takes a few seconds and then pops up the references dialog, same as when Go To References is used.

VS Code About infos:

Version: 1.50.1 (user setup)
Commit: d2e414d9e4239a252d1ab117bd7067f125afd80a
Date: 2020-10-13T15:06:15.712Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.18363

**Comment by mooware** [comment]
In the supposed duplicate issue linked here, I found the setting "editor.gotoLocation.alternativeDefinitionCommand" and with that the change log that explains the behavior:

> VS Code used to do nothing when selecting Go to Definition while already at the definition. With this release, alternate commands can be executed instead. For instance, Go to Definition can be mapped to run Go to References.

So apparently this is intentional. But this behavior seems extremely unintuitive to me. As a user, I might not be aware that I'm already inside the definition, or I might want to jump to the beginning of it. Instead of either simply performing the requested action, or notifying me that I'm already inside the definition, it does nothing or performs a different action that I did not request.
